src/Bridge/Graphql/Type/SchemaBuilder.php

```diff
@@ -0,0 +1,298 @@
+     * @return Schema
+     */
+    public function getSchema(): Schema
+    {
+        $queryFields = [];
+
+        foreach ($this->resourceNameCollectionFactory->create() as $resource) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($resource);
+
+            $queryFields = array_merge($queryFields, $this->getQueryFields($resource, $resourceMetadata));
+        }
+
+        return new Schema([
+            'query' => new ObjectType([
+                'name' => 'Query',
+                'fields' => $queryFields,
+            ]),
+        ]);
+    }
+
+    /**
+     * Gets the query fields of the schema.
+     *
+     * @param string           $resource
+     * @param ResourceMetadata $resourceMetadata
+     *
+     * @return array
+     */
+    private function getQueryFields(string $resource, ResourceMetadata $resourceMetadata): array
+    {
+        $queryFields = [];
+        $shortName = $resourceMetadata->getShortName();
+        $resourceType = $this->getResourceType($resource, $resourceMetadata);
+
+        foreach ($this->getOperations($resourceMetadata, true, true) as $operationName => $queryItemOperation) {
+            $fieldNamePrefix = 'get' === $operationName ? '' : $operationName;
+            $queryFields[lcfirst($fieldNamePrefix.$shortName)] = [
+                'type' => $resourceType,
+                'args' => ['id' => $this->getResourceIdentifierFieldArguments($resource)],
+                // @TODO Create a schema resolver class?
+                'resolve' => function ($root, $args, $context, ResolveInfo $resolveInfo) use ($resource, $operationName) {
+                    $item = $this->itemDataProvider->getItem($resource, $args['id'], $operationName);
+                    return $this->normalizer->normalize($item, null, ['graphql_selected_fields' => $resolveInfo->getFieldSelection()]);
+                },
+            ];
+        }
+
+        foreach ($this->getOperations($resourceMetadata, true, false) as $operationName => $queryCollectionOperation) {
+            $fieldNamePrefix = 'get' === $operationName ? '' : $operationName;
+            $queryFields[lcfirst($fieldNamePrefix.Inflector::pluralize($shortName))] = [
+                'type' => GraphQLType::listOf($resourceType),
+                // @TODO Create a schema resolver class?
+                'args' => [
+                    'first' => [
+                        'type' => GraphQLType::int(),
+                        'description' => 'Returns the first n elements from the list.'
+                    ],
+                ],
+                'resolve' => function ($root, $args, $context, ResolveInfo $resolveInfo) use ($resource, $operationName) {
+                    // @TODO Pagination
+                    $collection = $this->collectionDataProvider->getCollection($resource, $operationName);
+                    return $this->normalizer->normalize($collection, null, ['graphql_selected_fields' => $resolveInfo->getFieldSelection()]);
```

Comment:
Cant' we use https://github.com/symfony/symfony/pull/18834 instead of `graphql_selected_fields`? I've introduced this option exactly for GraphQL support :)

Reply:
I can replace `graphql_selected_fields` by `attributes`, yes. But it will not change the initial issue: to replace the generated URI (call to `normalizeRelation`) by the subresource data.

Reply:
I'm not even sure this is a good idea. If I replace it, it will have consequences when using `attributes` in an SQL context. See its use here: https://github.com/alanpoulain/core/blob/2ed4e8c54ac8579857fc401ad32b18ece0f962a4/src/Serializer/AbstractItemNormalizer.php#L416

Reply:
Ok, keep it as is for now :)

Reply:
Using `attributes` is not useful in this context because the fields are filtered by graphql-php. Maybe it can improves performance though?

Reply:
It will definitely improve performances, serialization is a costly operation, we should definitely prevent serializing useless data.